### DISCUSSION TITLE
How does the plugin work?

### DISCUSSION THREAD

**Discussion by HSDSZ** [bnpr/Malt#251] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/36077492/154424069-36ba8cfc-6562-4e0d-80df-d57590ec7e09.png)

![image](https://user-images.githubusercontent.com/36077492/154423771-ea096214-be94-441e-87f7-3cd3fb78e77d.png)

I added the plugin folder in the addon preference. But in the node panel, I still cannot find the node such as the instance_random_color

Blender 3.01, the latest Malt development version

**Comment by pragma37** [comment, marked as answer]
Take a look at https://github.com/bnpr/Malt/discussions/237

**Reply by HSDSZ** [reply]
It works. Thanks

**Comment by HSDSZ** [comment]
By choosing its parent folder, the plugin works. 
Then I have another question. If I want a make a custom node that based on the diffuse_gradient function. What should I do. I include the "NPR_Mesh.glsl", but it does not work
![image](https://user-images.githubusercontent.com/36077492/154644457-2759ea3a-606a-47d6-8963-a8872dc92853.png)

**Reply by pragma37** [reply]
Several things:

1. You can import glsl files relative to the Shader directory, so in this case:
```GLSL
#include "NPR_Pipeline/NPR_Mesh.glsl"
```
2. The example plugin loads the glsl library for all graph types (Mesh, Screen and Light). You would have to put the function on a separate library:
```Python
graphs['Mesh'].add_library(mesh_library_path)
```
Or put the includes and the functions inside a preprocessor conditional:
```GLSL
#ifdef IS_MESH_SHADER
// Mesh only code goes here
#endif
```

**Reply by HSDSZ** [reply]
Thanks. It works.
